main.js

```diff
@@ -312,15 +312,56 @@ function continueInit() {
       postGroup(token, userId, 'sequenceiq.account.' + userId + '.' + company)
   }
 
-  updateCloudbreakGroups = function (token, userId) {
-      updateGroup(token, userId, 'cloudbreak.templates')
-      updateGroup(token, userId, 'cloudbreak.stacks')
-      updateGroup(token, userId, 'cloudbreak.blueprints')
-      updateGroup(token, userId, 'cloudbreak.credentials')
-      updateGroup(token, userId, 'cloudbreak.events')
-      updateGroup(token, userId, 'cloudbreak.recipes')
-      updateGroup(token, userId, 'cloudbreak.usages.user')
-      updateGroup(token, userId, 'periscope.cluster')
+  updateCloudbreakGroups = function (token, userId, requestedScopes) {
+      if (requestedScopes == undefined || requestedScopes == null) {
+          updateGroup(token, userId, 'cloudbreak.templates')
+          updateGroup(token, userId, 'cloudbreak.stacks')
+          updateGroup(token, userId, 'cloudbreak.blueprints')
+          updateGroup(token, userId, 'cloudbreak.credentials')
+          updateGroup(token, userId, 'cloudbreak.events')
+          updateGroup(token, userId, 'cloudbreak.recipes')
+          updateGroup(token, userId, 'cloudbreak.usages.user')
+          updateGroup(token, userId, 'periscope.cluster')
+      } else {
+          if (requestedScopes.blueprints.write) {
```

Comment:
I guess is it possible to iterate on requestedScopes instead of hard coding opportunities. If it's about security, also possible to get available scopes from an array.